File: frontend/src/types/company.ts
export interface Company {
  id: string
  name: string
  industry?: string
  website?: string
  phone?: string
  email?: string
  address?: string
  city?: string
  country?: string
  postalCode?: string
  source?: string
  employeeCount?: number
  annualRevenue?: number
  notes?: string
  isActive: boolean
  createdAt: string
  updatedAt: string
}

export interface CreateCompanyDto {
  name: string
  industry?: string
  website?: string
  phone?: string
  email?: string
  address?: string
  city?: string
  country?: string
  postalCode?: string
  source?: string
  employeeCount?: number
  annualRevenue?: number
  notes?: string
  isActive?: boolean
}

export interface UpdateCompanyDto {
  name?: string
  industry?: string
  website?: string
  phone?: string
  email?: string
  address?: string
  city?: string
  country?: string
  postalCode?: string
  source?: string
  employeeCount?: number
  annualRevenue?: number
  notes?: string
  isActive?: boolean
}

export interface CompaniesResponse {
  items: Company[]
  totalCount: number
  pageNumber: number
  pageSize: number
  totalPages: number
}

File: backend/CrmApp.Domain/Entities/Company.cs
namespace CrmApp.Domain.Entities;

public class Company : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string? Industry { get; set; }
    public string? Website { get; set; }
    public string? Phone { get; set; }
    public string? Email { get; set; }
    public string? Address { get; set; }
    public string? City { get; set; }
    public string? Country { get; set; }
    public string? PostalCode { get; set; }
    public string? Source { get; set; } // Website, Referral, Cold Call, etc.
    public int? EmployeeCount { get; set; }
    public decimal? AnnualRevenue { get; set; }
    public string? Notes { get; set; }
    
    // Navigation properties
    public ICollection<Contact> Contacts { get; set; } = new List<Contact>();
    public ICollection<Lead> Leads { get; set; } = new List<Lead>();
    public ICollection<Opportunity> Opportunities { get; set; } = new List<Opportunity>();
}
